lib/connectors/memory.js

```diff
@@ -228,6 +228,15 @@ Memory.prototype.updateOrCreate = function (model, data, callback) {
   });
 };
 
+Memory.prototype.findOrCreate = function (model, query, data, callback) {
+  this.all(model, query, function (err, list) {
+    if (err || (list && list[0])) return callback(err, list && list[0], false);
+    this.create(model, data, function (err) {
+      callback(err, data, true);
+    });
+  }.bind(this));
+};
+
```

Comment:
Isn't is better to leave this out and let juggler to use the default unoptimized implementation for the memory connector?